### PR TITLE
workflows: Improve CI workflow

### PR DESCRIPTION
**Pull request by gthao313**
Add linting check, Builds and tests to workflows.

**Issue number:**
#36
#37

**Description of changes:**
```
Author: Tianhao Geng <tianhg@amazon.com>
Date:   Tue May 25 01:42:20 2021 +0000

workflows: Improve CI workflow

Add linting check, Builds and tests to workflows.
```

**Testing done:**
See this PR and [this actions run from the branch](https://github.com/bottlerocket-os/bottlerocket-update-operator/actions/runs/876034500)

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,8 +5,20 @@ on:
     branches:
       - develop
   pull_request:
+    branches:
+      - develop
```
Comment (samuelkarp) [inline .github/workflows/ci.yml]: You can remove this and it should run on pull requests to any branch.

Slightly more canonically, you can change it like this:

```yaml
on:
  - push:
      branches:
        - develop
  - pull_request
```

Or even better might be to get rid of the branch in `push` too:

```yaml
on:
  - push
  - pull_request
```

**Comment by gthao313** [comment]
Push above fix lint error and update workflow according to Sam's suggestion.

**Review by samuelkarp — APPROVED** [review]
LGTM

**Review by jahkeup — APPROVED** [review]
👍🏽 👏🏽 :shipit:

**Pull request merged**